File: RAI/backend/app/models/food.py
from dataclasses import dataclass
from datetime import datetime
from typing import Literal

FoodSource = Literal["manual", "open_food_facts", "internal"]


@dataclass(frozen=True)
class Food:
    id: str
    name: str

    brand: str | None = None
    barcode: str | None = None
    category: str | None = None

    calories_per_100g: float | None = None

    protein_g_per_100g: float | None = None
    carbs_g_per_100g: float | None = None
    fat_g_per_100g: float | None = None

    fiber_g_per_100g: float | None = None
    sugar_g_per_100g: float | None = None
    salt_g_per_100g: float | None = None

    source: FoodSource = "manual"
    source_id: str | None = None
    image_url: str | None = None

    is_verified: bool = False

    created_at: datetime | None = None
    updated_at: datetime | None = None

File: RAI/frontend/mobile/types/food.ts
export type FoodSource = "manual" | "open_food_facts" | "internal"

export type Food = {
    id: string;
    name: string;
    brand: string | null;
    barcode: string | null;
    category: string | null;
    calories_per_100g: 0 | null;
    protein_g_per_100g: 0 | null;
    carbs_g_per_100g: 0 | null;
    fat_g_per_100g: 0 | null;
    fiber_per_100g: 0 | null;
    sugar_per_100g: 0 | null;
    salt_per_100g: 0 | null;
    source: FoodSource;
    source_id: string | null;
    image_url: string | null;
    is_verified: boolean;
    created_at: string;
    updated_at: string;
}

export type FoodSearchParams = {
    query: string;
    limit?: number;
    skip?: number;
}

export type CreateFoodRequest = {
  name: string;
  brand?: string;
  calories_per_100g: number;
  protein_g_per_100g: number;
  carbs_g_per_100g: number;
  fat_g_per_100g: number;
};